### ISSUE TITLE
What is singularityhub?

### ISSUE THREAD

**Issue by YPatois** [singularityhub/singularityhub.github.io#102]
Hi,
Just starting with singularity.
The doc seems to miss some "what is singularityhub made for?"
I guess it's a way to store singularity images description and helping to instantiate it, but I dont know.
The doc start by "providing more detail on how to build, manage, and deploy your images", but well, before the details, what is it about?
Thanks.

**Comment by vsoch** [comment]
hey @YPatois ! I'd be glad to update this for you! Singularity Hub is a free an openly available registry for Singularity containers. It will build your containers from recipe files (simple text files) on Github, and then make them programmatically available with the Singularity software. Are you familiar with Singularity itself?

**Comment by vsoch** [comment]
Did you see the README, and was this missing information for you?

https://github.com/singularityhub/singularityhub.github.io#singularity-hub

**Comment by YPatois** [comment]
Thanks :-)
I didnt saw the README, that's right.
I went to look for the doc before having a look at the code in git.
Maybe a line in the doc could help anyway?
Thanks again.

**Comment by vsoch** [comment]
Sure thing! What would you suggest?

**Comment by YPatois** [comment]
Hi,
What you have in the readme looks good:
"What is Singularity Hub?

Singularity Hub is a registry for scientific images. The core of the Hub are the Singularity container images. A Singularity container image is built from a build file called Singularity, which is akin to the Dockerfile."

I'm not a native English speaker, so I may miss some details, anyway:

"scientific images" seems too vague, especially if one doesn't know much about containers and such (it's not a database of pictures).

Maybe "Singularity Hub is a registry for scientific container images to be used by Singularity."
Although "used" is pretty vague.

"registry" seems to be the right word, even if I dont understand it fully yet in the context.

Next, I would take the Singularity own words "Singularity containers can be used to package entire scientific workflows, software and libraries, and even data."

"Singularity Hub is a registry of  scientific Singularity container images. Singularity containers can be used to package entire scientific workflows, software and libraries, and even data."

I would drop the " a build file called Singularity"reference: it can me renamed, and anyway it's an implementation detail that doesnt fit here.

"Singularity Hub is a registry of  scientific Singularity container images. Singularity containers can be used to package entire scientific workflows, software and libraries, and even data. Singularity Hub provides tools to to build, manage and instantiate Singularity container images from its description files."

I would rephrase a bit the Docker reference. You cant assume people will know it, and wont make people think they have to know Docker to understand Singularity (which "akin Docker" seems to imply). However, as many know docker, a reference is useful:

"Singularity Hub is a registry of  scientific Singularity container images. Singularity containers can be used to package entire scientific workflows, software and libraries, and even data. Singularity Hub provides tools to remotely build, manage and instantiate Singularity container images from its description files, that can also manage Docker images if desired"

Not too happy with that last part, but couldn't find better. Anyway, the one who doesn't knows docker will understand she doesn't "desire" docker" and the one who knows will understand that Singularity is close to docker and even that she can reuse stuff she did with it.

Thanks for your attention.

**Comment by vsoch** [comment]
thanks @YPatois ! I just rewrote the first part of the README, let me know what you think. All of your points were spot on minus the detail for the build recipe. It must actually be called `Singularity` minimally (and can have some extension to indicate a tag) for the hub to search the repository and find it.

**Comment by YPatois** [comment]
Hi,
Thanks for your work, I like it :-)

But my original comment was about something else.
If I google "singularity hub", I wont come on the github page, but there:
https://singularity-hub.org/

And there I read : "Hi friends! Welcome to the new Singularity Hub. Read the Release Notes for important changes to the platform, browse our updated documentation , and please let us know if you run into any issues."

OK, as I dont know what is "singularity hub", I try the "documentation" link, and I get:
"News
Singularity 2.4 is released, and so is Singularity Hub 2.0! [...]"

OK, this next title seems more promising, "getting started!":
"Getting Started
This is the official user guide for singularity-hub.org, providing more detail on how to build, manage, and deploy your images."
... Well, not really. I still cant know what is Singularity hub...

If I'm supposed to go to the code repository to understand what this software is about, I think many wont ever find it...

So that definition should be more visible somewhere else. WHat do you think?

Thanks.
Yannick

**Comment by vsoch** [comment]
Ah I completely agree, the flow is terrible! (I hadn't thought about coming from the site). How about this fix?

https://github.com/singularityhub/singularityhub.github.io/wiki

So now you go from Singularity Hub website --> the link above, and it presents the more gentle opening. What do you think?

**Comment by vsoch** [comment]
Thanks for greatly improving this! I'm going to close the issue give all our work, and please open up again for discussion if you find any other unclear documentation flows.

**Comment by YPatois** [comment]
Great :-) Thanks again!
